src/Yaapii.Atoms/Number/LiveNumber.cs

```diff
@@ -0,0 +1,262 @@
+﻿// MIT License
+//
+// Copyright(c) 2019 ICARUS Consulting GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Globalization;
+using Yaapii.Atoms.Scalar;
+using Yaapii.Atoms.Texts;
+
+namespace Yaapii.Atoms.Number
+{
+    /// <summary>
+    /// A parsed number
+    /// </summary>
+    public sealed class LiveNumber : INumber
+    {
+        private readonly IScalar<long> _lng;
+        private readonly IScalar<float> _flt;
+        private readonly IScalar<int> _itg;
+        private readonly IScalar<double> _dbl;
+
+        /// <summary>
+        /// A <see cref="IText"/> as a <see cref="INumber"/>
+        /// </summary>
+        /// <param name="text">text to parse</param>
+        /// <param name="blockSeperator">seperator for blocks, for example 1.000</param>
+        /// <param name="decimalSeperator">seperator for floating point numbers, for example 16,235 </param>
+        public LiveNumber(Func<string> text, string decimalSeperator, string blockSeperator) : this(
+            new LiveScalar<long>(() => Convert.ToInt64(
```

Comment:
I suggest you just use the NumberOf object here in combination with a func which creates it every time.